src/ImageSharp/Formats/Webp/WebpDecoderCore.cs

```diff
@@ -440,6 +447,38 @@ private void ParseOptionalExtendedChunks(WebpChunkType chunkType, WebpFeatures f
 
                     break;
 
+                case WebpChunkType.Exif:
+                    uint exifChunkSize = this.ReadChunkSize();
+                    if (this.IgnoreMetadata)
+                    {
+                        this.currentStream.Skip((int)exifChunkSize);
+                    }
+                    else
+                    {
+                        byte[] exifData = new byte[exifChunkSize];
+                        this.currentStream.Read(exifData, 0, (int)exifChunkSize);
```

Comment:
Stream.Read doesn't guarantee that the requested length is actually read. It may be less.
Is that case covered? (on other places as well)

Reply:
It is handled by `BufferedReadStream` stream wrapper:

https://github.com/SixLabors/ImageSharp/blob/fe044d131c45d89aa52281e2eb442c82b4424ce1/src/ImageSharp/Formats/Webp/WebpDecoderCore.cs#L81

In actual `Read` methods:

https://github.com/SixLabors/ImageSharp/blob/fe044d131c45d89aa52281e2eb442c82b4424ce1/src/ImageSharp/IO/BufferedReadStream.cs#L342-L352

Shouldn't we also save stream reference as explicit `BufferedReadStream` instead of abstract `Stream` for de-virtualization?

Reply:
> Shouldn't we also save stream reference as explicit `BufferedReadStream` instead of abstract `Stream` for de-virtualization?

Yep that would be better.
JIT's dynamic PGO (profile guided optimization) could handle this, but right now this is opt-in by setting the environment varialbe `export DOTNET_TieredPGO=1` in the host-process.

Besides that it would be clear that's a `BufferedReadStream` and not the base `Stream`, so concerns about reading less than the requested lenght won't arise 😉.